src/pages/Regions/RegionEdit.tsx

```diff
@@ -99,32 +100,50 @@ const RegionEditFormPage = () => {
   };
 
   const handleSubmit = async (values: any) => {
-    const newRegion: any = {
-      regionName: values.regionName,
-      description: values.description,
-      phoneNumber: values.phoneNumber,
-      email: values.email,
-      link: values.link,
-      street: values.street,
-      houseNumber: values.houseNumber,
-      officeNumber: values.officeNumber,
-      postIndex: values.postIndex,
-      logo: logo,
-      city: values.city,
-      isActive: chosenRegion.isActive
-    };
-    await RegionsApi.EditRegion(currentRegion, newRegion);
+    setLoading(true);
 
-    form.resetFields();
+    if (!await checkIfNameExists(values.regionName)) {
+      const newRegion: RegionProfile = {
+        id: chosenRegion.id,
+        regionName: values.regionName,
+        description: values.description,
+        phoneNumber: values.phoneNumber,
+        email: values.email,
+        link: values.link,
+        street: values.street,
+        houseNumber: values.houseNumber,
+        officeNumber: values.officeNumber,
+        postIndex: values.postIndex,
+        logo: logo,
+        city: values.city,
+        isActive: chosenRegion.isActive
+      };
+      await RegionsApi.EditRegion(currentRegion, newRegion);
+  
+      form.resetFields();
+  
+      notificationLogic("success", successfulEditAction("Дані округи"));
+      history.push(`/regions/${currentRegion}`);  
+    } else {
+      setLoading(false);
 
-    notificationLogic("success", successfulEditAction("Дані округи"));
-    history.push(`/regions/${currentRegion}`);
+      showRegionNameExistsModal();
+    }
   };
 
+  function showRegionNameExistsModal() {
```

Comment:
You have exact the same method in `AddRegion.tsx`
You could move it to some common class for them both

Reply:
we use methods of this type only to call the necessary modal windows in each component separately, so I don't think that it is necessary to move them to a separate class

Reply:
as you wish
but it still code duplicate😃

Reply:
Removed code dublicate